test/quantization/fx/test_quantize_fx.py

```diff
@@ -3269,24 +3269,70 @@ def __init__(self):
             def forward(self, x):
                 return self.linear(x)
 
-        m = M().eval()
-        m = prepare_fx(m, {"": default_qconfig})
-        m_ref = copy.deepcopy(m)
-        m_ref = convert_fx(m_ref, is_reference=True)
-        m = convert_fx(m)
-        data = torch.randn(8, 5)
-        out_ref = m_ref(data)
-        out = m(data)
-        # check that reference pattern for quantized linear module is fused
-        expected_node_occurrence = {
-            ns.call_function(torch.quantize_per_tensor): 1,
-            ns.call_module(torch.nn.quantized.Linear): 1,
-            ns.call_method("dequantize"): 1
-        }
-        self.checkGraphModuleNodes(m, expected_node_occurrence=expected_node_occurrence)
+        def _test(qat):
+            if qat:
+                m = M()
+                m = prepare_qat_fx(m, {"": default_qat_qconfig})
+            else:
+                m = M().eval()
+                m = prepare_fx(m, {"": default_qconfig})
+            m_ref = copy.deepcopy(m)
+            m_ref = convert_fx(m_ref, is_reference=True)
+            m = convert_fx(m)
+            data = torch.randn(8, 5)
+            out_ref = m_ref(data)
+            out = m(data)
+            # check that reference pattern for quantized linear module is fused
+            expected_node_occurrence = {
+                ns.call_function(torch.quantize_per_tensor): 1,
+                ns.call_module(nnq.Linear): 1,
+                ns.call_method("dequantize"): 1
+            }
+            self.checkGraphModuleNodes(m, expected_node_occurrence=expected_node_occurrence)
+
+            # checking result match
+            self.assertEqual(out_ref, out)
+
+        _test(qat=False)
+        _test(qat=True)
+
+    def test_linear_relu_lowering(self):
```

Comment:
we do have some existing tests that already tests this path. maybe another thing we can do is to clean up those tests a bit:
https://github.com/pytorch/pytorch/blob/master/test/quantization/fx/test_quantize_fx.py#L3609

Reply:
I see. I guess the main difference is this test explicitly compares the lowered and non-lowered models. I just modeled it after `test_linear_lowering`. I wonder if it makes sense to merge these two tests?

Reply:
As discussed offline, we will try to merge this test with `test_linear_module`